api/v1/helpers/verbs/doGet.js

```diff
@@ -29,11 +31,41 @@ const httpStatus = require('../../constants').httpStatus;
  *  resource type to retrieve.
  */
 function doGet(req, res, next, props) {
-  u.findByKey(props, req.swagger.params)
-  .then((o) => {
-    res.status(httpStatus.OK).json(u.responsify(o, props, req.method));
-  })
-  .catch((err) => u.handleError(next, err, props.modelName));
+  if (props.cacheEnabled) {
+    const reqParams = req.swagger.params;
+    let cacheKey = reqParams.key.value;
+
+    // cache key is combination of key and fields in specified order.
```

Comment:
I don't think it will hurt us to set the cache key to include the list of fields in alphabetical order, but I'm also OK with leaving this as is for now (i.e. accepting that there may be fewer cache hits) and getting performance numbers later to see if the extra time to alphabetize the  field list would have any impact vs. getting a higher cache hit rate.